pkg/storage/action_cache_asset_store.go

```diff
@@ -104,11 +106,23 @@ func (rs *actionCacheAssetStore) Get(ctx context.Context, ref *asset.AssetRefere
 	if err != nil {
 		return nil, err
 	}
-	// Create an action using the asset ref and directory containing
-	// qualifiers
-	action, _, err := assetReferenceToAction(ref, directoryDigest)
-	if err != nil {
-		return nil, err
+	var action *remoteexecution.Action
+	if commandGenerator, err := qualifier.QualifiersToCommand(ref.Qualifiers); err != nil || len(ref.Uris) > 1 {
+		// Create the action with the qualifier directory as the input root
+		action, _, err = assetReferenceToAction(ref, directoryDigest)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		command := commandGenerator(ref.Uris[0])
+		commandDigest, err := ProtoToDigest(command)
+		if err != nil {
+			return nil, err
+		}
+		action = &remoteexecution.Action{
+			CommandDigest:   commandDigest,
+			InputRootDigest: EmptyDigest,
```

Comment:
This is incorrect, should be `directoryDigest` not `EmptyDigest`. Just adding this comment to make sure I remember to change it.